src/lib/storage/value_column.cpp

```diff
@@ -155,6 +155,37 @@ std::shared_ptr<BaseColumn> ValueColumn<T>::copy_using_allocator(const Polymorph
   }
 }
 
+template <typename T>
+MemoryUsage ValueColumn<T>::estimate_memory_usage(MemoryUsageEstimationMode estimation_mode) const {
+  return MemoryUsage{sizeof(*this) + _values.size() * sizeof(T) +
+                     (_null_values ? _null_values->size() * sizeof(bool) : 0u)};
+}
+
+template <>
+MemoryUsage ValueColumn<std::string>::estimate_memory_usage(MemoryUsageEstimationMode estimation_mode) const {
+  auto bytes = size_t{(_null_values ? _null_values->size() * sizeof(bool) : 0u) + sizeof(*this)};
+
+  switch (estimation_mode) {
+    case MemoryUsageEstimationMode::Fast:
+      return MemoryUsage{bytes + _values.size() * sizeof(std::string)};
```

Comment:
Actually, this is just for the particular case of strings in value columns. Fast will just assume they are all SSO, MoreExact will look at each individual string and reason whether SSO might get used. It might still be wrong. 

I agree this isn't cool. Just iterating over all rows (there might be billions of them!!) for a quick memory estimation might not be what the user wants.

Remove one mode? If so, which?

Reply:
Once we actually deal with [billions and biilions](https://www.youtube.com/watch?v=u_aLESDql1U) of rows, we hopefully have the tracking allocator.

For now, I don't think we need the fast (and incorrect) mode.